### ISSUE TITLE
Losing ALL tabs when using multiple windows

### ISSUE THREAD

**Issue by xanthos84** [zen-browser/desktop#2471]
### Captchas

- [X] I have read the instructions.
- [X] I have searched existing issues and avoided creating duplicates.
- [X] I am not filing an enhancement request.

### What happened?

I don't know if I'm in the right place, but for me this behavior is a headache - as I always have to worry about losing all my tabs (across all workspaces!!) when working with multiple windows open.

Especially when I want to completely restart Zen, for example, I have to be very careful which window is closed first.

To make the whole thing understandable, I have made a screen recording where I first open a window with PINNED tabs and then a second window, then close the first again - then close the second. And restart Zen, the first window with all its tabs (and tabs in the workspaces!) is lost.

Recording: https://imgur.com/gG5sWGg

Is this a bug? Or intended? IF intended, why? Shouldn't we have persistant windows/tabs/workspaces?

### Reproducible?

- [ ] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

1.0.1a16

### What platform are you seeing the problem on?

Windows

**Comment by adithyaharun** [comment]
+1 happened to me as well.

- Open a window with pinned tabs and recently opened tabs.
- Open a new window, with same profile. Boom, tabs gone, pinned tabs also gone.
- Close both window, and
- Voila! Open the new window and you got your profile's tabs wiped out.

**Comment by anatasiajp** [comment]
When using multiple windows, use Quit menu to save all session

**Comment by salamijason** [comment]
Yes this is an ongoing issue and it has to do with session store. What firefox does is takes the most recent active window and saves it as your last "session". Because of this, when you're closing one window at a time, you're effectively resetting your session over and over again with a new set of tabs. 

Persistent pinned tabs are being worked on by the team but have not been released.

Best way to avoid this is using the quite function like the above user mentioned. It's found at the end of the three dot menu: ![Image](https://github.com/user-attachments/assets/aa8d3ef7-ec81-4d37-aae3-c6cd71dd383f)

**Comment by mbledkowski** [comment]
This is actually quite frustrating of an issue for me. I would love to see it being fixed on the release

**Comment by ryouzue** [comment]
Reviving this issue 'cus it persists, I've just confirmed today on **1.7.4b (Firefox 134.0.2) (64-bit)** after downloading for the first time.
I'll write a step-by-step to replicate this bug
```
> Open Zen browser
> Open an extra tab
> Split one of the tabs out of the main window

> You should have 2 windows with 1 zen tab open on each
> Close one of either windows
> Tada 🎉 You successfully lost all the tabs you had open!
```
Replicating this bug will erase EVERY tab you had open in ALL workspaces, even if you got "Open previous windows and tabs" option enabled in Settings.

_tl:dr; I don't know what the devs are thinking marking this as **Completed** even though the bug is still out there in the wild_

> **Half-baked suggestion from me is**; Create a temporary workspace named like so, for the tab that was dragged out of the main window and persistently keep saving the rest of the tabs open

**Comment by ryouzue** [comment]
Reviving this issue 'cus it persists, I've just confirmed today on **1.7.4b (Firefox 134.0.2) (64-bit)** after downloading for the first time.
I'll write a step-by-step to replicate this bug
```
> Open Zen browser
> Open an extra tab
> Split one of the tabs out of the main window

> You should have 2 windows with 1 zen tab open on each
> Close one of either windows
> Reopen a window by right clicking on task bar
> Tada 🎉 You successfully lost all the tabs you had open on the one you closed!
```
Replicating this bug will erase EVERY tab you had open in ALL workspaces UNLESS they are pinned (...what), even if you got "Open previous windows and tabs" option enabled in Settings.

_tl:dr; I don't know what the devs are thinking marking this as **Completed** even though the bug is still out there in the wild_

> **Half-baked suggestion from me is**; Create a temporary workspace named like so, for the tab that was dragged out of the main window and persistently keep saving the rest of the tabs open

> **Tl:dr;**; Just move the tab that was dragged out of the main window into a temporary workspace

**Comment by mr-cheffy** [comment]
Session restore only remembers your last window unless all are closed at the same time.

**Comment by ryouzue** [comment]
Wouldn't it be possible to make session memorize all the closed windows so the next time zen reopens, it'll load both of them including the right window sizes?

**Comment by Kagukara** [comment]
Just had zen-browser delete all my workspace tabs for the 3rd time. I'm 100% sure I keep the window with them open at all times but I guess not.

Why not have it like how vivaldi or floorp does workspaces? All windows  share the same workspaces but if one window has workspace "a" open the other windows cannot access workspace "a" etc etc.

Also adding a way to save the current workspaces and all the tabs within them would be a life saver at the moment with the tabs being deleted and all.

**Comment by mr-cheffy** [comment]
## Closing this issue as part of a repo cleanup

Due to technical-debt we had in the past, we are currently clearing out old, duplicate, stale, and non-actionable issues to improve issue tracking and make it easier to focus on fixing real bugs.

If this issue is still relevant, please open a new issue using our updated issue template, which will help us categorize and address it more efficiently.

Thanks for your understanding and for helping us improve the project! 🚀

This issue can be reopened at: https://github.com/zen-browser/desktop/issues/new?template=bug_report.yml&title=Losing%20ALL%20tabs%20when%20using%20multiple%20windows%0A
